app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,32 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    # Ensure all account endpoints are active/configured
+    account.switch do
+      cleanup_solr(account)
+      cleanup_fedora(account)
+      # Redis doesn't need 'account', as switch changes Redis::Namespace to correct namespace
+      cleanup_redis
```

Comment:
OK, I see your point.  Go ahead with whatever arrangement you prefer.  With your larger switch block, it might make sense to change `cleanup_fedora`'s param to just be the base path though, since it does nothing with account anymore.